gd1_helpers/membership/gd1_model.py

```diff
@@ -77,19 +91,143 @@ class GD1BackgroundModel(GD1Base, StreamModel):
         "pm2": {"x": "phi1", "y": "pm2", "y_err": "pm2_err"},
     }
 
+    @classmethod
+    def bkg_update(cls, pawprint, data, knot_sep):
```

Comment:
Hm, in my original vision for the structure, updating the class parameters should just happen statically on the class itself. Do you use this heavily?

Reply:
I tried for a long time to figure out how to do everything without having a function like this but was not able to figure it out and was wasting a ton of time on it so decided to do this. I know it wasn't automatically updating the way I needed to the way it was before.
Also, as I was writing the code, I was trying to make it generalizable. It's probably easier to make it much more compact if written only for GD-1.